redfish-core/lib/log_services.hpp

```diff
@@ -6873,36 +6974,27 @@ inline void fillSystemHardwareIsolationLogEntry(
                         messages::internalError(asyncResp->res);
                         break;
                     }
-
-                    if (*severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Critical")
-                    {
-                        entryJson["Severity"] = "Critical";
-                    }
-                    else if (
-                        *severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Warning")
-                    {
-                        entryJson["Severity"] = "Warning";
-                    }
-                    else if (
-                        *severity ==
-                        "xyz.openbmc_project.HardwareIsolation.Entry.Type.Manual")
+                    const std::string& severityString = *severity;
+                    guardType =
+                        severityString.substr(severityString.rfind('.') + 1);
+                    entryJson["Severity"] = guardType;
+                    // Manual and Spare guard type map to severity "OK"
+                    if (severityString ==
+                            "xyz.openbmc_project.HardwareIsolation.Entry.Type.Manual" ||
+                        severityString ==
+                            "xyz.openbmc_project.HardwareIsolation.Entry.Type.Spare")
                     {
                         entryJson["Severity"] = "OK";
                     }
-                    else
-                    {
-                        BMCWEB_LOG_ERROR(
-                            "Unsupported Severity[{}] from object: {}",
-                            *severity, dbusObjIt->first.str);
-                        messages::internalError(asyncResp->res);
-                        break;
```

Comment:
did we miss this code?

Reply:
Yeah. We did not want to keep this code. Talked to Raviteja and this is an overhead everytime we add a new guard type, we need to add a if condition here. As this is an enum, we could not have other values.